### PR TITLE
Feat/block chart edit

### PR DESCRIPTION
**Pull request by preda-bogdan**
### Summary

Added a button inside Gutenberg to Edit the Chart from a single location. Will open a Media pop-up Modal, the same that is used outside of the editor.
Added a new filter `visualizer_legacy_block_edit` to allow users to switch to the legacy behavior. Eg.
```php
add_filter( 'visualizer_legacy_block_edit', '__return_true' );
```
Hide previous controls from the block.
Added E2E test.

### Will affect the visual aspect of the product

YES

### Screenshots 
<details open>
    <summary>Edit Chart</summary>

image
</details>

<details>
    <summary>Media Pop-up</summary>

image
</details>

### Test instructions

1. All tests should pass.
2. On a fresh instance add a chart inside the Gutenberg editor.
3. Check that the Edit Chart button is present and previous controls are not available.
4. Check that. the button will open the Media pop-up and allows you to edit the current chart.

## Check before Pull Request is ready:

* [x] I have [written a test](CONTRIBUTING.md#writing-an-acceptance-test) and included it in this PR
* [x] I have [run all tests](CONTRIBUTING.md#run-tests) and they pass
* [x] The code passes when [running the PHP CodeSniffer](CONTRIBUTING.md#run-php-codesniffer)
* [x] Code meets [WordPress Coding Standards](CONTRIBUTING.md#coding-standards) for PHP, HTML, CSS and JS
* [x] [Security and Sanitization](CONTRIBUTING.md#security-and-sanitization) requirements have been followed
* [x] I have assigned a reviewer or two to review this PR (if you're not sure who to assign, we can do this step for you)

Closes Codeinwp/visualizer-pro#440.

### CODE REVIEW

File: classes/Visualizer/Gutenberg/src/Components/ChartSelect.js [hunk 1]
```diff
@@ -32,13 +32,70 @@ const { startCase } = lodash;
...
 const {
 	Component,
 	Fragment
 } = wp.element;
 
 const { InspectorControls } = wp.blockEditor || wp.editor;
 
+let vm, vmv;
+vm = visualizer.media = {};
+vmv = vm.view = {};
```
Comment (Soare-Robert-Daniel) [inline classes/Visualizer/Gutenberg/src/Components/ChartSelect.js:48]: We can use names like `visualizerMediaView` to improve readability. We not have to keep the awful names from `library.js`

File: classes/Visualizer/Gutenberg/src/Components/ChartSelect.js [hunk 2]
```diff
@@ -80,53 +138,80 @@ class ChartSelect extends Component {
             footer = __( 'Annotations in this chart may not display here but they will display in the front end.' );
         }
 
+		const openEditChart = ( chartId ) => {
+			let wnd = window;
+			const baseURL = ( wnd.visualizerLocalize.chartEditUrl ) ? wnd.visualizerLocalize.chartEditUrl : '';
```
Comment (Soare-Robert-Daniel) [inline classes/Visualizer/Gutenberg/src/Components/ChartSelect.js:143]: I see no use for a rename of `window` to `wnd`.

**Comment by irinelenache** [comment]
@preda-bogdan Tested and found a single issue:

- [ ] If you edit the chart using the pop-up, there will be a console error after saving and the changes will be visible after refreshing the editor https://vertis.d.pr/v/wsi8xB

Can be checked here:
```
 Admin area URL: https://block-chart.s3-tastewp.com/wp-admin 
 Username: irinel 
 Password: v9M2uK9vP2Q 
```

**Comment by Soare-Robert-Daniel** [comment]
> * [ ]  If you edit the chart using the pop-up, there will be a console error after saving, and the changes will be visible after refreshing the editor https://vertis.d.pr/v/wsi8xB

@preda-bogdan you can add:

```php
window.send_to_editor = () => {
    view.close();
};
```

I found the same issue yesterday.

**Comment by preda-bogdan** [comment]
@Soare-Robert-Daniel thank you, but we also need to notify the editor to re-render the block.

**Comment by preda-bogdan** [comment]
@irinelenache The chart should update upon the window closing and no error should be displayed in the console.

Let me know if everything is working as expected now. Thank you!

**Comment by irinelenache** [comment]
@preda-bogdan The issue is fixed now, thank you 🚀

**Pull request merged**